binding.c

```diff
@@ -523,78 +862,245 @@ NAPI_METHOD(udx_napi_socket_bind) {
 
   // wont error in practice
   err = udx_socket_recv_start(self, on_udx_message);
-  if (err < 0) UDX_NAPI_THROW(err)
+  if (err < 0) {
+    napi_throw_error(env, uv_err_name(err), uv_strerror(err));
+    return NULL;
+  }
 
-  NAPI_RETURN_UINT32(local_port)
+  napi_value return_uint32;
+  if (napi_create_uint32(env, local_port, &return_uint32) != napi_ok) {
+    napi_throw_error(env, NULL, "napi_create_uint32(env, local_port, &return_uint32) failed!");
+    return NULL;
+  }
+
+  return return_uint32;
 }
 
-NAPI_METHOD(udx_napi_socket_set_ttl) {
-  NAPI_ARGV(2)
-  NAPI_ARGV_BUFFER_CAST(udx_socket_t *, self, 0)
-  NAPI_ARGV_UINT32(ttl, 1)
+napi_value
+udx_napi_socket_set_ttl (napi_env env, napi_callback_info info) {
+  napi_value argv[2];
+  size_t argc = 2;
+  if (napi_get_cb_info(env, info, &argc, argv, NULL, NULL) != napi_ok) {
+    napi_throw_error(env, NULL, "napi_get_cb_info(env, info, &argc, argv, NULL, NULL) failed!");
+    return NULL;
+  }
+
+  udx_socket_t *self;
+  size_t self_len;
+  if (napi_get_buffer_info(env, argv[0], (void **) &self, &self_len) != napi_ok) {
+    napi_throw_error(env, NULL, "napi_get_buffer_info(env, argv[0], (void **) &self, &self_len) failed!");
+    return NULL;
+  }
+
+  uint32_t ttl;
+  if (napi_get_value_uint32(env, argv[1], &ttl) != napi_ok) {
+    napi_throw_error(env, "EINVAL", "Expected unsigned number");
+    return NULL;
+  }
 
   int err = udx_socket_set_ttl(self, ttl);
-  if (err < 0) UDX_NAPI_THROW(err)
+  if (err < 0) {
+    napi_throw_error(env, uv_err_name(err), uv_strerror(err));
```

Comment:
for clarity, keep error handling liek this one - thats a real error that can happen